chain-signatures/node/src/protocol/presignature.rs

```diff
@@ -558,6 +578,20 @@ impl PresignatureManager {
                         break;
                     }
                     Action::SendMany(data) => {
+                        if generator.poked_latest.is_none() {
+                            let now = Instant::now();
+                            let start_time = generator.timestamp;
+                            presignature_before_poke_delay_metric
+                                .observe((now - start_time).as_secs_f64());
+                            generator.poked_latest = Some((now, Duration::from_millis(0), 1));
+                        } else {
+                            let (last_poked, total_wait, total_pokes) =
+                                generator.poked_latest.unwrap();
+                            let now = Instant::now();
+                            let elapsed = now - last_poked;
+                            generator.poked_latest =
+                                Some((now, total_wait + elapsed, total_pokes + 1));
+                        }
```

Comment:
should probably make this into its own function since its used quite a bit, and also should do a `if let Some(val)` instead of checking on `is_none` especially since we're using the values inside of the option in one of the branches anyways. Also, why are we doing `Instant::now - last_poked()` when we could just do `last_poked.elapsed()`
```suggestion
                        let (total_wait, total_pokes) = if let Some((last_poked, total_wait, total_pokes)) = &generator.poked_latest {
                            (total_wait + last_poked.elapsed(), total_pokes + 1)
                        } else {
                            let start_time = generator.timestamp;
                            presignature_before_poke_delay_metric
                                .observe(start_time.elapsed().as_secs_f64());
                            (Duration::from_millis(0), 1)
                        };
                        let last_poked = Instant::now();
                        generator.poked_latest = Some((last_poked, total_wait, total_pokes));
```